### PR TITLE
Gracefully handle /api errors

### PR DESCRIPTION
**Pull request by mmcgahan**
CSRF errors, oauth errors, etc

The main culprit is the `onPreResponse` handler that will generate an HTML response when there's an error in the request/response to `/api`, which the app will choke on because it's not JSON.

### CODE REVIEW

File: src/apiProxy/apiProxyRoutes.js [hunk 1]
```diff
@@ -27,12 +27,22 @@ const getApiProxyRoutes = (path, env, apiProxyFn$) => {
 	 *   by `apiAdapter.apiResponseToQueryResponse`
 	 */
 	const handler = getApiProxyRouteHandler(proxyApiRequest$);
+	const plugins = {
+		'electrode-csrf-jwt': {
+			enabled: true,
+		}
+	};
```
Comment (mmcgahan) [inline src/apiProxy/apiProxyRoutes.js]: this was accidentally removed in a previous merge

**Comment by mmcgahan** [comment]
I want to check the route config `ext` to see if i can add a custom `onPreResponse` that will override the HTML-generating one. That might be cleaner than the `jsonError` config

**Comment by mmcgahan** [comment]
okay all set - API_ERROR actions now add a `state.app.error` value that the consumer app can use to do something in the UI

File: src/routes/appRoute.js [hunk 2]
```diff
@@ -1,12 +1,39 @@
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
 import { getAppRouteHandler } from './appRouteHandler';
 
 /**
+ * This function provides global error handling when there is a 500 error
+ */
+export const onPreResponse = {
```
Comment (mmcgahan) [inline src/routes/appRoute.js]: this was moved from being a global `onPreResponse` handler to being app-route-specific

File: src/util/tracking.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ export const trackApi = log => (response, queryResponses, metadata={}) => {
 	}
 	// special case - login requests need to be tracked
 	const loginResponse = queryResponses.find(r => r.login);
-	if (loginResponse) {
+	if ((loginResponse && loginResponse.login.value || {}).member) {
```
Comment (sadafie) [inline src/util/tracking.js]: is `loginResponse.login` ever undefined here?

Comment (mmcgahan) [inline src/util/tracking.js]: no - the `.find(r => r.login)` will only return something that has a truthy `.login` property

Comment (sadafie) [inline src/util/tracking.js]: cool

**Pull request merged**